kivy/network/urlrequest.py

```diff
@@ -633,26 +555,239 @@ def cancel(self):
+            ctx.check_hostname = False
+            ctx.verify_mode = ssl.CERT_NONE
+            args['context'] = ctx
+
+        if self._proxy_host:
+            Logger.debug('UrlRequest: {0} - proxy via {1}:{2}'.format(
+                id(self), self._proxy_host, self._proxy_port
+            ))
+            req = cls(self._proxy_host, self._proxy_port, **args)
+            if parse.scheme == 'https':
+                req.set_tunnel(host, port, self._proxy_headers)
+            else:
+                path = urlunparse(parse)
+        else:
+            req = cls(host, port, **args)
+
+        # send request
+        method = self._method
+        if method is None:
+            method = 'GET' if body is None else 'POST'
+
+        req.request(method, path, body, headers or {})
+
+        # read header
+        return req, req.getresponse()
+
+
+class UrlRequestRequests(UrlRequestBase):
+
+    def get_chunks(
+            self, resp, chunk_size, total_size, report_progress, q,
+            trigger, fd=None
+    ):
+        bytes_so_far = 0
+        result = b''
+
+        for chunk in resp.iter_content(chunk_size):
+            if not chunk:
+                break
+
+            if fd:
+                fd.write(chunk)
+
+            else:
+                result += chunk
+
+            bytes_so_far += len(chunk)
+
+            if report_progress:
+                q(('progress', resp, (bytes_so_far, total_size)))
+                trigger()
+
+            if self._cancel_event.is_set():
+                break
+
+        return bytes_so_far, result
+
+    def get_response(self, resp):
+        return resp.content
+
+    def get_total_size(self, resp):
+        return int(resp.headers.get('Content-Length', -1))
+
+    def get_content_type(self, resp):
+        return resp.headers.get('Content-Type', None)
+
+    def get_status_code(self, resp):
+        return resp.status_code
+
+    def get_all_headers(self, resp):
+        return resp.headers.items()
+
+    def close_connection(self, req):
+        pass
+
+    def call_request(self, body, headers):
+        timeout = self._timeout
+        ca_file = self.ca_file
+        verify = self.verify
+        url = self._requested_url
+        auth = self._auth
+
+        req = requests
+        kwargs = {}
+
+        # get method
+        if self._method is None:
+            method = 'get' if body is None else 'post'
+        else:
+            method = self._method.lower()
+
+        req_call = getattr(req, method)
+
+        if auth:
+            kwargs["auth"] = auth
+
+        # send request
+        response = req_call(
+            url,
+            data=body,
+            headers=headers,
+            timeout=timeout,
+            verify=verify,
+            cert=ca_file,
+            **kwargs
+        )
 
-    def on_error(req, error):
-        pprint('Got an error:')
-        pprint(error)
+        return None, response
 
-    Clock.start_clock()
```

Comment:
i dont know what is this part, without it still works well [even tests]

Reply:
then it should be fine to remove, if it breaks something, at least we'll know why we need it :)